thorntail-runner/src/main/java/org/wildfly/swarm/runner/WarBuilder.java

```diff
@@ -159,32 +159,29 @@ private void addClassesToZip(ZipOutputStream output, String prefix, File classes
         try {
             Files.walk(classesDirectory.toPath())
                     .map(Path::toFile)
-                    .filter(File::isFile)
                     .forEach(file -> addClassToZip(output, prefix, file, classesDirectory));
         } catch (IOException e) {
             throw new RuntimeException("Failed to add classes to war", e);
         }
     }
 
     private void addClassToZip(ZipOutputStream output, String prefix, File file, File classesDirectory) {
+        URI base = classesDirectory.toURI();
         try {
-            String name = getRelativePath(file, classesDirectory.getAbsolutePath());
+            String name = base.relativize(file.toURI()).getPath();
             name = prefix + name;
-            writeFileToZip(output, file, name);
+            if (file.isDirectory()) {
```

Comment:
Was it failing for you without adding an entry for directory? 
It seems it worked without it so far.

Reply:
Yes it was failing for me without an entry for a folder. I don't know about linux system, but on windows it was failing.

Reply:
thanks for clarification